Source/Scene/Model.js

```diff
@@ -2533,6 +2707,89 @@ define([
+                var a = accessors[accessorId];
+                var extensions = a.extensions;
+
+                if (defined(extensions)) {
+                    var quantizedAttributes = extensions.WEB3D_quantized_attributes;
+                    if (defined(quantizedAttributes)) {
+                        var decodeMatrix = quantizedAttributes.decodeMatrix;
+                        var uniformVariable = 'czm_u_dec_' + attribute.toLowerCase();
+
+                        switch (a.type) {
+                            case 'VEC2':
+                                uniformMap[uniformVariable] = getMat3UniformFunction(decodeMatrix, model).func;
+                                setUniforms.push(uniformVariable);
+                                break;
+                            case 'VEC3':
+                                uniformMap[uniformVariable] = getMat4UniformFunction(decodeMatrix, model).func;
+                                setUniforms.push(uniformVariable);
+                                break;
+                            case 'VEC4':
+                                // VEC4 attributes are split into scale and translate because there is no mat5 in GLSL
+                                var uniformVariableScale = uniformVariable + '_scale';
+                                var uniformVariableTranslate = uniformVariable + '_translate';
+                                uniformMap[uniformVariableScale] = getMat4UniformFunction(scaleFromMatrix5Array(decodeMatrix), model).func;
+                                uniformMap[uniformVariableTranslate] = getVec4UniformFunction(translateFromMatrix5Array(decodeMatrix), model).func;
+                                setUniforms.push(uniformVariableScale);
+                                setUniforms.push(uniformVariableTranslate);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+
+        // If there are any unset quantized uniforms in this program, they should be set to the identity
+        for (var quantizedUniform in quantizedUniforms) {
+            if (quantizedUniforms.hasOwnProperty(quantizedUniform)) {
+                if (setUniforms.indexOf(quantizedUniform) < 0) {
+                    var properties = quantizedUniforms[quantizedUniform];
+                    if (defined(properties.mat)) {
+                        switch (properties.mat) {
+                            case 3:
+                                uniformMap[quantizedUniform] = getMat3UniformFunction(Matrix3.IDENTITY, model).func;
+                                break;
+                            case 4:
+                                uniformMap[quantizedUniform] = getMat4UniformFunction(Matrix4.IDENTITY, model).func;
+                                break;
+                        }
+                    }
+                    if (defined(properties.vec)) {
+                        switch (properties.vec) {
```

Comment:
Here and above, switch these to `if` statements (no pun intended).  It will be more concise.  If it were a longer list, `switch` would be appropriate.